Sources/Common/Extensions/UIKitExtensions.swift

```diff
@@ -10,4 +10,14 @@ public extension UIApplication {
     }
 }
 
+public extension UIViewController {
+    var moduleNameOfView: String? {
+        Bundle(for: type(of: self)).bundleIdentifier
+    }
+
+    var isSwiftUIView: Bool {
+        moduleNameOfView == "com.apple.SwiftUI"
```

Comment:
I think this is true considering, all SwiftUI view controllers are subclasses of UIHostingController only then they would have `bundleIdentifier` equal to "com.apple.SwiftUI". Can there be a case where this is not true ?

Reply:
This is checking what module a class belongs to. That means that any class that is in the SwiftUI framework will have this same module name. 

That's why I like this solution and am confident in it - it's based on module names (this wouldn't change on each iOS release) instead of something more fragile like a class name (apple could rename "UIHostingController", for example).